device/futhark/CMakeLists.txt

```diff
@@ -0,0 +1,140 @@
+# TRACCC library, part of the ACTS project (R&D line)
+#
+# (c) 2021-2022 CERN for the benefit of the ACTS project
+#
+# Mozilla Public License Version 2.0
+
+# The C language is not enabled in our project by default, but we need it to
+# compile Futhark's code.
+enable_language(C)
+
+# Try to find our dependencies, which are of course Futhark itself (required)
+# and the CUDA toolkit for compiling CUDA projects. Also notify the user of
+# what we find.
+find_program(Futhark futhark REQUIRED)
+find_package(CUDAToolkit)
+
+message(STATUS "Found Futhark compiler at ${Futhark}")
+
+# This function describes how to turn a Futhark source into a CMake library.
+# The function creates a target called `TARGET_NAME` from a source file
+# `_SOURCE_NAME`. The Futhark target platform (CUDA or C) is given
+# (case-insensitively) by `_LANGUAGE_TARGET`
+function(add_futhark_library TARGET_NAME _LANGUAGE_TARGET _SOURCE_NAME)
```

Comment:
I don't like this interface. :frowning: I don't like that we're forced into building a separate **shared** library out of every Futhark file.

I can be on board with having a dedicated "Futhark library" creating function, which can have regular C\+\+ code as well along with the specific "Futhark code".

But I would much rather have a function like `add_futhark_sources_to_target(...)` that would add the generation of sources for a specific, already existing target.

Would that be possible to do? Of course the goal would be to allow to add multiple "Futhark sources", in multiple steps for this. Like:

```cmake
traccc_add_library( Foo ... )
add_futhark_source_to_target( Foo bar1.fut ... )
add_futhark_source_to_target( Foo bar2.fut ... )
```